File: 프로그래머스/lv1/42862. 체육복/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int count = 0;
        int[] arr = new int[n+2];
        for(int l:lost){
            arr[l]++;
        }
        for(int r:reserve){
            arr[r]--;
        }
        for(int i=0; i<arr.length; i++){
            if(arr[i] == -1){
                if(arr[i-1] == 1){
                    arr[i]++;
                    arr[i-1]--;
                } else if(arr[i+1] == 1){
                    arr[i]++;
                    arr[i+1]--;
                }
            }
        }
        for(int i:arr){
            if(i == 1){
                count++;
            }
        }
        answer = n-count;
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    overlap = set(lost) & set(reserve)
    lost = [x for x in lost if x not in overlap]
    reserve = [x for x in reserve if x not in overlap]
    answer = n - len(lost)
    
    for i in range(1, n+1):
        if i in lost:
            if i-1 in reserve:
                reserve.remove(i-1)
                lost.remove(i)
                answer += 1
                continue
            elif i+1 in reserve:
                reserve.remove(i+1)
                lost.remove(i)
                answer += 1
                continue
    
    return answer